[Term entry] Python Binascii Module: .a2b_uu() (#4553)
Codecademy/docs: content/python/concepts/binascii-module/terms/a2b-uu/a2b-uu.md

---
Title: '.a2b_uu()'

> Description: 'Returns a new object with decoded binary data from ASCII-encoded data in the UUEncode format.'

Reviewer: ```suggestion
Description: 'Returns a new object containing the decoded binary data from ASCII-encoded data in the UUEncode format.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:

>   - 'ASCII'
>   - 'Unicode'
>   - 'UUEncode'

Reviewer: ```suggestion
  - 'Encoding'
  - 'Functions'
  - 'Unicode'
```

CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

> The `.a2b_uu()` function converts a single line of UUEncoded data back to binary and return the binary data.

Reviewer: ```suggestion
The **`.a2b_uu()`** function converts a single line of UUEncoded data back to binary and returns the binary data.
```

## Syntax

```pseudo
binascii.a2b_uu(string)
```

> The function accepts Unicode strings containing only ASCII characters.

Reviewer: ```suggestion
The `string` parameter represents the UUEncoded data intended for decoding back into binary format. Specifically, it should be a Unicode string containing only ASCII characters and conforming to the UUEncode format.
```

> ## Codebyte Example

Reviewer: ```suggestion
## Example
```

> ```codebyte/python
> import binascii
> 
> # Example ASCII-encoded data in uuencode format
> encoded_data = '*0V]D96-A9&5M>0  \n'
> 
> # Decode the ASCII-encoded data
> decoded_data = binascii.a2b_uu(encoded_data)
> 
> print(decoded_data)
> # Displays the decoded message
> ```

Reviewer: ```suggestion
```python
import binascii

# Example ASCII-encoded data in uuencode format
encoded_data = '*0V]D96-A9&5M>0  \n'

# Decode the ASCII-encoded data
decoded_data = binascii.a2b_uu(encoded_data)

# Displays the decoded message
print(decoded_data)
```
```

> 

Reviewer: ```suggestion
The example produces the following output:

```shell
b'Codecademy'
```
```

> 

Reviewer: This extra line can be removed.

---
Title: '.a2b_uu()'
Description: 'Returns a new object containing the decoded binary data from ASCII-encoded data in the UUEncode format.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Encoding'
  - 'Functions'
  - 'Unicode'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'
---

The **`.a2b_uu()`** function converts a single line of UUEncoded data back to binary and returns the binary data.

## Syntax

```pseudo
binascii.a2b_uu(string)
```
The `string` parameter represents the UUEncoded data intended for decoding back into binary format. Specifically, it should be a Unicode string containing only ASCII characters and conforming to the UUEncode format.

## Example

```python
import binascii

# Example ASCII-encoded data in uuencode format
encoded_data = '*0V]D96-A9&5M>0  \n'

# Decode the ASCII-encoded data
decoded_data = binascii.a2b_uu(encoded_data)

> print(decoded_data)
> # Displays the decoded message

Reviewer: ```suggestion
# Displays the decoded message
print(decoded_data)
```

```

The example produces the following output:

> ```shell
> b'Codecademy'

Reviewer: ```suggestion
```shell
b'Codecademy'
```
```

Author: @mamtawardhani, is there a change suggested here? Thanks!

Reviewer: @DIVL7 yes it should be: (The ending backticks are missing)

```shell
b'Codecademy
```
